### PR TITLE
Handle courses with missing start_at date

### PR DESCRIPTION
**Pull request by cycomachead**
This happens sometimes for CS88.
* Try to request the date of the *term* if present
* If no dates, fallback to created_at which should always be present

### CODE REVIEW

File: server/services/canvas/__init__.py [hunk 1]
```diff
@@ -32,25 +32,43 @@ def get_user(canvas_id, key=None) -> FakeUser | User:
...
+    # Ensure a valid start_at date for a course.
+    # return the term start_at_date if present
+    if course['term'] and course['term']['start_at']:
+        start_at = course['term']['start_at']
+        start_at_date = course['term']['start_at_date']
+    else:
+        start_at_date = course['start_at_date'] or course['created_at_date']
+        start_at = course['start_at'] or course['created_at']
+    course['start_at_date'] = start_at_date
+    course['start_at'] = start_at
```
Comment (Reimirno) [inline server/services/canvas/__init__.py:56]: wait shouldn't we use dot notation 
`course.start_at` instead `course['start_at']`

File: server/services/canvas/__init__.py [hunk 2]
```diff
@@ -32,25 +32,43 @@ def get_user(canvas_id, key=None) -> FakeUser | User:
...
 
 
 def get_user_courses_categorized(user: FakeUser | User) \
         -> tuple[list[FakeCourse | Course], list[FakeCourse | Course], list[FakeCourse | Course]]:
-    courses_raw = user.get_courses(enrollment_status='active')
+    courses_raw = user.get_courses(enrollment_status='active', include=['term'], per_page=100)
```
Comment (Reimirno) [inline server/services/canvas/__init__.py:61]: would need to change fake canvas client to support `include` keyword

Comment (cycomachead) [reply server/services/canvas/__init__.py:61]: oh does it not? I thought it did.. :( 
https://canvasapi.readthedocs.io/en/stable/keyword-args.html?highlight=include#list-parameters

File: server/services/canvas/__init__.py (hunk 1 above)
Comment (cycomachead) [reply server/services/canvas/__init__.py:56]: oh sure...I wasn't sure, I thought these were dicts, but yeah maybe we should... I assume a.b.c is valid syntax then? I don't remember all the rules of Python or if we need to check for presence before an error is realized..

File: server/services/canvas/__init__.py (hunk 2 above)
Comment (Reimirno) [reply server/services/canvas/__init__.py:61]: The real one does.
Our fake one does not.
I will do it.

File: server/services/canvas/__init__.py (hunk 1 above)
Comment (Reimirno) [reply server/services/canvas/__init__.py:56]: No worries I will take it from here.

**Pull request merged**